### ISSUE TITLE
Talk Proposal: Grammar of Graphics and Functional Programming

### ISSUE THREAD

**Issue by adrianmroz** [berlinjs/berlinjs.org#285]
# Grammar of Graphics and Functional Programming in Data Visualisation”

### Abstract

Sometimes libraries with charts are not enough - they lack customisation or don’t support that one chart type you would like to render. You can always create a custom graph and I’d like to show the audience basics and tips on how to create such custom data visualisations.

Grammar of Graphics gives us a great framework and mental model for creating charts. We can learn how to encode data on screen using marks, scales and channels. All these abstractions could be implemented easily using UI components and functions.

The presentation would be targeted at data visualisation novices and I would use React for examples on slides, but these techniques are applicable to all common, component-based frameworks - I used this approach in React and Angular with great success.

The Audience will learn about Grammar of Graphics and obtain tools to create great custom visualisations. It will enable them to create impactful charts when common libraries won’t suffice.

### About me

I’m a frontend developer who has worked with data visualisations for 5 years
and I've been a web developer for 8 years. Currently, I work for Allegro as the main maintainer of data exploration tool turnilo: github.com/allegro/turnilo

### Details

It will be my first talk in English. My previous talks in polish are [here](https://github.com/meetjspl/poznan/issues?utf8=%E2%9C%93&q=is%3Aissue+author%3Aadrianmroz) with slides and few with recordings. The talk will take 20-25 minutes.

**Comment by carolstran** [comment]
That's great to hear! We're on the third Thursday of every month - as of now, I believe that every month except June and July are possibilities 😊

**Comment by carolstran** [comment]
Hey, just wanted to follow-up.

Here are the meetup dates for the rest of the year (except July and September because we already have a full lineup). Let me know if any of these work for you 😊

August 15
October 17
November 21

**Comment by adrianmroz** [comment]
I’d like to call dibs on October 17. I will confirm later this week - I need to handle paperwork in my company.

Can’t wait! :)

**Comment by adrianmroz** [comment]
@carolstran I Died you at berlinjs slack, let's discuss details there, shall we?

**Comment by carolstran** [comment]
Ooooh ok I haven't logged in there for a while. I'll do that after work 😊

**Comment by carolstran** [comment]
Finally messaged you back I'm the worst 🙈 😭 Sorry about the delay!!!! But to iterate what I said: We'd still love to have you in October if you're available 😁

**Comment by carolstran** [comment]
Hey hey @adrianmroz, just wanted to check in and see if you're still up for giving a talk at our meetup later this month (October 17) 👻  And if yes, then please let me know if you want to make any changes to your title/abstract before we add it to the website 🎃

**Comment by adrianmroz** [comment]
Yup, I'll be there. I'll give you shorter abstract today - current one is quite dry :D

**Comment by carolstran** [comment]
Awesome! Can't wait 🎉 

Once I get confirmation from the other speakers I'll update the website, post on Twitter, etc. Aiming for next week 🕺

**Comment by adrianmroz** [comment]
Sorry for late response!

---

### Foundations of custom data visualisations.

Sometimes libraries with charts are not enough - they lack customisation or don’t support that one chart type you would like to use.  I'd like to show you how to start with custom data visualisations using grammar of graphics. Grammar of graphics is a popular model of graphs and charts. We can combine it with functional programming to produce nice data visualizations.

**Comment by carolstran** [comment]
Great, so I've updated the website - just let me know if we should change anything: https://berlinjs.org/ Planning to announce on Twitter tomorrow!

**Comment by carolstran** [comment]
Do you have a Twitter account by the way? I linked your GitHub on the site because I wasn't sure

**Comment by adrianmroz** [comment]
Looks good. No, I don’t have public Twitter account ;)

**Comment by adrianmroz** [comment]
I couldn't find mail for organisers team or some better channel. I can't come tomorrow. I know it is very late but reason is sudden and very serious and I'm sorry. I would like to come to your meet up eventually in the future if you let me. If you need more information from me, please reach me via EMAIL (I'm not comfortable sharing more information via public issue tracker)

**Comment by rmehner** [comment]
@adrianmroz Ugh, very sorry to hear and don't worry, life is more important than our meetup. No need to share more details.

All the best and let us know if we can help with anything!

**Comment by carolstran** [comment]
I was just writing (almost exactly) the same thing.

Wishing you all the best and know that you're always welcome to come and speak in the future.

**Comment by rmehner** [comment]
FYI: I've updated your comment and removed your email address, no need to leak that.

For future reference, you can always email me directly (robin@coding-robin.de) or us (info@berlinjs.org) if you want. Sending good vibes!
